dotCMS/src/main/java/com/dotmarketing/portlets/folders/action/OrderMenuAction.java

```diff
@@ -109,30 +108,30 @@ public void processAction(ActionMapping mapping, ActionForm form, PortletConfig
 					_sendToReferral(req,res,req.getParameter("referer"));
 					return;
 				}
-				
-                // we have to clear navs after db commit
-                for(Treeable treeable : navs){
-                    Identifier id = APILocator.getIdentifierAPI().find(treeable.getIdentifier());
-                    if("folder".equals(id.getAssetType())){
-                        CacheLocator.getNavToolCache().removeNavByPath(id.getHostId(), id.getPath());
-                    }
-                    
-                    Folder folder = APILocator.getFolderAPI().findParentFolder(treeable, user, false);
-                    String folderInode = (folder==null) ? FolderAPI.SYSTEM_FOLDER : folder.getInode();
-                    CacheLocator.getNavToolCache().removeNav(id.getHostId(), folderInode);
-                    CacheLocator.getHTMLPageCache().remove(id.getId());
-                    CacheLocator.getNavToolCache().removeNav(id.getHostId(), FolderAPI.SYSTEM_FOLDER);
-                }
-
-
-                
-                
-                
 				_sendToReferral(req,res,"/html/portlet/ext/folders/redirect_after_order.jsp");
-				
-				
 
-				
+
+				// we have to clear navs after db commit
+				for(Treeable treeable : navs){
+					Identifier id = APILocator.getIdentifierAPI().find(treeable.getIdentifier());
+					if("folder".equals(id.getAssetType())){
+						CacheLocator.getNavToolCache().removeNavByPath(id.getHostId(), id.getPath());
+					}
+
+					Folder folder = APILocator.getFolderAPI().findParentFolder(treeable, user, false);
+					String folderInode = (folder==null) ? FolderAPI.SYSTEM_FOLDER : folder.getInode();
+					CacheLocator.getNavToolCache().removeNav(id.getHostId(), folderInode);

```

Comment:
Shouldn't these nav invalidations be happening in the apis?

Reply:
they should yes